### ISSUE TITLE
Allow rewind of debugger statements to rerun previous lines

### ISSUE THREAD

**Issue by cgchoffman** [Komodo/KomodoEdit#351]
Basically rewinding the stack a step.  Allows the user to execute a line during runtime, then rerun this line (during runtime) with modified inputs.

I can see this being handy in large applications where you have no choice but to wait for a lot of code to run to reach a breakpoint you've set.  I've hit this with a network application where faking the data (for final cut purposes) and bypassing certain code wasn't an option so had to wait several minutes for initialization steps to occur.

**Comment by ericpromislow** [comment]
Reversible debugging is available for languages like C and Java. But in the
languages we support, no. How are you going to reset the interpreter's program counter?
AFAIK, none of our supported languages expose it.

**Comment by Naatan** [comment]
Right, sadly not possible :\ Although I guess we could store information on different lines to show you "what happened there?"

So you wouldn't be able to walk back down the stack, but you'd be able to glance at what was happening at that point in time, eg. variable definitions.
